react-app/src/app/components/RestrictRoute/index.js

```diff
@@ -3,12 +3,16 @@ import { Route, Redirect } from 'react-router-dom';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 
-import { getUser } from '../../../redux/Auth/actions';
-
 class RestrictRoute extends Component {
   render() {
-    this.props.getUser(this.props.token);
-    if (this.props.logged) {
+    // TODO: where should this be done?
+    // this.props.getUser(this.props.token);
+    if (this.props.logged && this.props.path === '/login') {
```

Comment:
If you had a signup screen, you would need to add it here. Maybe you can do this same logic when the route is not private and the user is logged, in order to do a generic solution for every public route